src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,243 @@
+        }
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String query = "UPDATE cars c set c.model = ? where c.id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        removeCarsDrivers(car);
+        insertCarsDrivers(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars set model = ?, manufacturer_id = ?"
+                + " where id = ? and deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, id);
+            return statementForCars.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: "
+                    + id + " into cars", exception);
+        }
     }
 
+    @Override
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "INNER JOIN cars_drivers cd ON cd.car_id = c.id "
+                + "INNER JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "INNER JOIN drivers d ON d.id = cd.driver_id "
+                + "WHERE cd.driver_id = ? AND c.deleted = FALSE AND d.deleted = FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getAllCarsFromResultSet(resultSet);
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+    }
```

Comment:
same as for getAll method: call createCar() and createManufacturer() inside the try block

call getDriversByCarId() after the catch block

don't use getAllCarsFromResultSet()